process/block/baseProcess.go

```diff
@@ -1239,3 +1242,66 @@ func (bp *baseProcessor) addHeaderIntoTrackerPool(nonce uint64, shardID uint32)
 		bp.blockTracker.AddTrackedHeader(headers[i], hashes[i])
 	}
 }
+
+func (bp *baseProcessor) commitTrieEpochRootHashIfNeeded(metaBlock *block.MetaBlock) error {
+	trieEpochRootHashStorageUnit := bp.store.GetStorer(dataRetriever.TrieEpochRootHashUnit)
+	if check.IfNil(trieEpochRootHashStorageUnit) {
+		return nil
+	}
+	_, isStorerDisabled := trieEpochRootHashStorageUnit.(*storageUnit.NilStorer)
+	if isStorerDisabled {
+		return nil
+	}
+
+	userAccountsDb := bp.accountsDB[state.UserAccountsState]
+	if userAccountsDb == nil {
+		return nil
+	}
+
+	currentRootHash, err := userAccountsDb.RootHash()
+	if err != nil {
+		return err
+	}
+	epochBytes := make([]byte, 4)
+	binary.BigEndian.PutUint32(epochBytes, metaBlock.Epoch)
+
+	err = trieEpochRootHashStorageUnit.Put(epochBytes, currentRootHash)
+	if err != nil {
+		return err
+	}
+
+	allLeaves, err := userAccountsDb.GetAllLeaves(currentRootHash)
+	if err != nil {
+		return err
+	}
+
+	balanceSum := big.NewInt(0)
+	for addressKey, userAccountsBytes := range allLeaves {
+		userAccount, errUnmarshal := unmarshalUserAccount([]byte(addressKey), userAccountsBytes, bp.marshalizer)
+		if errUnmarshal != nil {
+			log.Debug("cannot unmarshal user account. it may be a code leaf", "error", errUnmarshal)
+			continue
+		}
+		balanceSum.Add(balanceSum, userAccount.GetBalance())
+	}
+
+	log.Debug("sum of addresses in shard at epoch start",
```

Comment:
since you compute the total balance, why not saving it in trieEpochRootHashStorageUnit (might rename it).

Reply:
i would leave it like this for the moment. after further discussions over the feature branch, we can rename the storer and use it in multiple ways